packages/astro/src/core/sync/index.ts

```diff
@@ -28,73 +22,92 @@ import {
 import type { Logger } from '../logger/core.js';
 import { formatErrorMessage } from '../messages.js';
 import { ensureProcessNodeEnv } from '../util.js';
-
-export type ProcessExit = 0 | 1;
+import { createNodeLogger } from '../config/logging.js';
+import { resolveConfig } from '../config/config.js';
+import { createSettings } from '../config/settings.js';
+import { telemetry } from '../../events/index.js';
+import { eventCliSession } from '../../events/session.js';
+import { runHookConfigSetup } from '../../integrations/hooks.js';
 
 export type SyncOptions = {
 	/**
 	 * @internal only used for testing
 	 */
 	fs?: typeof fsMod;
-};
-
-export type SyncInternalOptions = SyncOptions & {
 	logger: Logger;
+	settings: AstroSettings;
+	skip?: {
+		// Must be skipped in dev
+		content?: boolean;
+	};
 };
 
 type DBPackage = {
 	typegen?: (args: Pick<AstroConfig, 'root' | 'integrations'>) => Promise<void>;
 };
 
+export async function syncInlineConfig({
+	inlineConfig,
+	fs,
+	telemetry: _telemetry = false,
+}: { inlineConfig: AstroInlineConfig; fs?: typeof fsMod; telemetry?: boolean }) {
+	const logger = createNodeLogger(inlineConfig);
+	const { astroConfig, userConfig } = await resolveConfig(inlineConfig ?? {}, 'sync');
+	if (_telemetry) {
+		telemetry.record(eventCliSession('sync', userConfig));
+	}
+	let settings = await createSettings(astroConfig, inlineConfig.root);
+	settings = await runHookConfigSetup({
+		command: 'build',
+		settings,
+		logger,
+	});
+	return await sync({ settings, logger, fs });
+}
```

Comment:
IIUC this is the main function to call from the programmatic API (`sync()`). In which case, I think this should be kept as `sync()` name and shouldn't be renamed as `syncInlineConfig()`.

The other default exported `sync()` function should be renamed as something else in that case. And should no longer be default exported since it's not the main `sync` function (the `syncInlineConfig` would be default exported).